modules/ocf/manifests/ssl/default.pp

```diff
@@ -0,0 +1,26 @@
+# Provides the key, certificate, and Let's Encrypt CA certificate bundle at
+# /etc/ssl/private/$cert_name.{key,crt,bundle,intermediate}
+class ocf::ssl::default {
+  # Attempt to collect all domains for a host to include in a SSL certificate.
+  # The '@' record needs to be handled in a special case, since
+  # "@.ocf.berkeley.edu" and "@.ocf.io" are both not valid domains
+  if '@' in $::dnsA {
+    $extra_domains = ['ocf.berkeley.edu', 'ocf.io']
+  } else {
+    $extra_domains = []
+  }
+
+  $domains = suffix(delete(concat(
+    [$::hostname],
+    delete($::dnsA, '@'),
+    $::dnsCname,
+  ), ''), '.ocf.berkeley.edu') + suffix(delete(concat(
+    [$::hostname],
+    delete($::dnsA, '@'),
+    $::dnsCname,
+  ), ''), '.ocf.io') + $extra_domains
```

Comment:
I think the only way this could be really deduplicated is to use [puppet functions](https://puppet.com/docs/puppet/5.3/lang_write_functions_in_puppet.html), I'll try that and see how it goes, but it might be too complicated for our use here, idk

Reply:
We can't assign the result of `delete(concat([$::hostname], delete($::dnsA, '@'), $::dnsCname), '')` to some variable?

Reply:
Sure, that would work too lol